packages/next/src/server/lib/router-server.ts

```diff
@@ -668,7 +668,15 @@ export async function initialize(opts: {
         // assetPrefix overrides basePath for HMR path
         if (assetPrefix) {
           hmrPrefix = normalizedAssetPrefix(assetPrefix)
+
+          if (URL.canParse(hmrPrefix)) {
+            // remove trailing slash from pathname
+            // return empty string if pathname is '/'
+            // to avoid conflicts with '/_next' below
+            hmrPrefix = new URL(hmrPrefix).pathname.replace(/\/$/, '')
+          }
         }
+
         const isHMRRequest = req.url.startsWith(
           ensureLeadingSlash(`${hmrPrefix}/_next/webpack-hmr`)
         )
```

Comment:
Previous Behavior:

The HMR was broken and content was not updated.
Because we check for `req.url` (which starts with `/path/name`), and it would never set `isHMRRequest` to true.
It is because if the `assetPrefix` is `http://localhost:3000`, `hmrPrefix` would also be `http://localhost:3000` and it can never match the pathname-style `req.url`